main/src/ca/uwaterloo/flix/language/phase/Parser.scala

```diff
@@ -1495,6 +1495,67 @@ class Parser(val source: Source) extends org.parboiled2.Parser {
 
   }
 
+  object Keywords {
+    def Absent: Rule0 = rule(keyword("Absent"))
```

Comment:
I would prefer if we capitalize all these defs. It is more consistent with the rest of the compiler and it avoids the use of escaped names, e.g. `case`. Otherwise, I think this looks great!

Reply:
I used lowercase names since parboiled's macro system uses the name of the def as the hint in the error message. If we use uppercase names here, we'll have `Def`, `Pub`, `Not`, etc. instead of `def`, `pub`, `not`. Will that be ok?

Reply:
Ah gosh-- then I don't like the setup at all :/

Reply:
I guess the use `keyword` in the first place is what prevents it from showing `def`.

Reply:
I think we can override the auto-naming with [`namedRule`](https://github.com/sirthias/parboiled2/blob/937c20bbad69913b8709afbe0eb44318b848f7da/parboiled-core/src/main/scala/org/parboiled2/Parser.scala#L43)

Then the change is ultimately a one-liner in the `keyword` definition.

Reply:
Cool. What does this do? From 200 lines to one line.

Reply:
Oh, it just uses the argument. That is nice.